File: app/src/main/java/ui/dialog/MoreDialog.java
package ui.dialog;

import android.app.AlertDialog;
import android.content.Context;
import android.view.LayoutInflater;

import hlv.cute.todo.R;
import hlv.cute.todo.databinding.DialogMoreBinding;

public class MoreDialog {

    private AlertDialog dialog;

    private OnClickEdit onClickEdit;
    private OnClickDetail onClickDetail;
    private OnClickDelete onClickDelete;

    public MoreDialog(Context context, boolean cancelable) {
        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.TranslucentDialog);
        builder.setCancelable(cancelable);
        DialogMoreBinding binding = DialogMoreBinding.inflate(LayoutInflater.from(context), null, false);
        builder.setView(binding.getRoot());

        binding.txtEdit.setOnClickListener(view -> {
            if (onClickEdit == null) {
                dismiss();
                return;
            }

            onClickEdit.onClick();
        });

        binding.txtDelete.setOnClickListener(view -> {
            if (onClickDelete == null) {
                dismiss();
                return;
            }

            onClickDelete.onClick();
        });

        binding.txtDetail.setOnClickListener(v -> {
            if (onClickDetail == null) {
                dismiss();
                return;
            }

            onClickDetail.onClick();
        });


        if (dialog == null)
            dialog = builder.create();
    }

    public MoreDialog(Context context) {
        this(context, true);
    }

    public void setOnClickEdit(OnClickEdit onClickEdit) {
        this.onClickEdit = onClickEdit;
    }

    public void setOnClickDetail(OnClickDetail onClickDetail) {
        this.onClickDetail = onClickDetail;
    }

    public void setOnClickDelete(OnClickDelete onClickDelete) {
        this.onClickDelete = onClickDelete;
    }

    public void dismiss() {
        if (dialog != null)
            dialog.dismiss();
    }

    public void show() {
        if (dialog != null)
            dialog.show();
    }

    public interface OnClickEdit {
        void onClick();
    }

    public interface OnClickDetail {
        void onClick();
    }

    public interface OnClickDelete {
        void onClick();

    }

}

File: app/src/main/java/ui/dialog/MoreDialog.kt
package ui.dialog

import android.app.AlertDialog
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import hlv.cute.todo.R
import hlv.cute.todo.databinding.DialogMoreBinding

class MoreDialog(context: Context?, cancelable: Boolean = true) {

    private var dialog: AlertDialog? = null

    private val binding: DialogMoreBinding

    var onClickEdit: (() -> Unit)? = null
    var onClickDetail: (() -> Unit)? = null
    var onClickDelete: (() -> Unit)? = null

    init {
        binding = DialogMoreBinding.inflate(LayoutInflater.from(context), null, false)

        val builder = AlertDialog.Builder(context, R.style.TranslucentDialog).apply {
            setCancelable(cancelable)
            setView(binding.root)
        }

        binding.txtEdit.setOnClickListener {
            if (onClickEdit == null) {
                dismiss()
                return@setOnClickListener
            }
            onClickEdit!!()
        }

        binding.txtDelete.setOnClickListener {
            if (onClickDelete == null) {
                dismiss()
                return@setOnClickListener
            }
            onClickDelete!!()
        }

        binding.txtDetail.setOnClickListener {
            if (onClickDetail == null) {
                dismiss()
                return@setOnClickListener
            }
            onClickDetail!!()
        }

        if (dialog == null)
            dialog = builder.create()
    }

    fun setWithDetail(withDetail: Boolean) {
        if (withDetail) {
            binding.txtDetail.visibility = View.VISIBLE
            binding.line2.visibility = View.VISIBLE
        } else {
            binding.txtDetail.visibility = View.GONE
            binding.line2.visibility = View.GONE
        }
    }

    fun show() = dialog?.show()

    fun dismiss() = dialog?.dismiss()
}